cmd/generate.go

```diff
@@ -81,6 +82,40 @@ func genProcessCluster(cmd *cobra.Command, clusterName string, p *ants.Pool) {
 		}
 	}
 
+	// list of components in cluster
+	for c, _ := range clusterComponents {
+		var clComps []string
+		clComps = append(clComps, c)
+
+		// only doing at cluster level as doing it in component process level was pain
+		if clusters == "" {
+			if components != "" {
+				break
+			}
+		}
+		d, err := os.Open(clusterDir)
+		if err != nil {
+			log.Fatal().Err(err).Msg("")
+		}
+		defer d.Close()
+		comDir, err := d.Readdirnames(-1)
+		if err != nil {
+			log.Fatal().Err(err).Msg("")
+		}
+		for _, dirName := range comDir {
```

Comment:
I find this hard to read. Essentially what you want is two slices/arrays, `comDir` which is all the components generated for a cluster, and `clComps` which should be the same as `clusterComponents` which is all the components assigned to the cluster. 
You want to find all the items in `comDir` that aren't in `clusterComponents` which would be easy if golang had sets :)

I think you want to split this into 3 pieces:
1. Find the set of components to be removed
2. Transform them from component name -> file path
3. delete the path

I think it'll look something like:

```
var unusedComponents = []string{}
for _, d := range comDir {
  for _, e :=  clComps {
    if e == d { break; } // found in both slices
  }
  unusedComponents = append(unusedComponents, d)
}
deleteComponents(unusedComponents)
```

Reply:
refactored the logic to use a map, which fixes the weird bug of deleting everything 😆